src/app/common/dialogs/export-pdf-dialog/export-pdf-dialog.component.ts

```diff
@@ -120,225 +191,280 @@ export class ExportPdfDialog extends BaseDialog {
+							this.doc.addImage(image, 'PNG', 0, 0, pageWidth, pageHeight, undefined, 'NONE');
+
+							//  OEB Logo
+							this.doc.addImage(oeb_logo, 'PNG', marginXImageLogo, yPos + 10, logoWidth, logoHeight);
+							this.doc.setDrawColor('#492E98');
+							this.doc.line(pageWidth / 2 - 50, 25, pageWidth / 2 + 100, 25);
+
+							if (esco) {
+								this.doc.textWithLink(
+									'* Kompetenz nach ESCO: https://esco.ec.europa.eu/de',
+									10,
+									pageHeight - 20,
+									{
+										url: 'https://esco.ec.europa.eu/de',
+									},
+								);
+							}
+
+							for (let i = 0; i < num_competencies; i++) {
+								if (i != 0 && i % competenciesPerPage === 0) {
+									this.doc.addPage();
+									this.doc.addImage(image, 'PNG', 0, 0, pageWidth, pageHeight, undefined, 'NONE');
+									//  OEB Logo
+									this.doc.addImage(
+										oeb_logo,
+										'PNG',
+										marginXImageLogo,
+										yPos + 10,
+										logoWidth,
+										logoHeight,
+									);
+									this.doc.setDrawColor('#492E98');
+									this.doc.line(pageWidth / 2 - 50, 25, pageWidth / 2 + 100, 25);
+								}
+
+								this.doc.setFontSize(28);
+								this.doc.setFont('Helvetica', 'bold');
+								this.doc.setTextColor('#492E98');
+								this.doc.text('Kompetenzen', 10, 60);
+
+								this.doc.setFontSize(18);
+								this.doc.setFont('Helvetica', 'normal');
+								this.doc.text('die ', 10, 75);
+								this.doc.setFont('Helvetica', 'bold');
+								this.doc.text(firstName + ' ' + lastName, this.doc.getTextWidth('die ') + 10, 75);
+								this.doc.setFont('Helvetica', 'normal');
+								const textWidth =
+									this.doc.getTextWidth('die ') +
+									15 +
+									this.doc.getTextWidth(firstName + ' ' + lastName);
+								this.doc.text('mit dem Badge', textWidth, 75);
+
+								this.doc.setFont('Helvetica', 'bold');
+								this.doc.text(badgeClass.name, 10, 85);
+
+								this.doc.setFont('Helvetica', 'normal');
+								this.doc.text('erworben hat:', this.doc.getTextWidth(badgeClass.name) + 15, 85);
+
+								let studyLoadInMin = competencies[i].studyLoad;
+								let studyLoadInHours = studyLoadInMin / 60;
+
+								const y = 110 + (i % competenciesPerPage) * 16;
+
+								this.doc.roundedRect(15, y - 7.5, 42.5, 11, 5, 5);
+								this.doc.setFontSize(14);
+								this.doc.addImage(dataUriClockPng, 20, y - 6, 12.5, 14);
+								if (studyLoadInMin > 60) {
+									this.doc.text(studyLoadInHours + ' Stunden', 30, y) + 2.5;
+								} else {
+									this.doc.text(studyLoadInMin + ' Minuten', 30, y) + 2.5;
+								}
+
+								this.doc.setFontSize(18);
+
+								if (competencies[i].escoID) {
+									this.doc.text(competencies[i].name + ' *', 60, y);
+								} else {
+									this.doc.text(competencies[i].name, 60, y);
+								}
+							}
+						}
+
+						this.doc.setPage(1);
+						this.doc.addImage(image, 'PNG', 0, 0, pageWidth, pageHeight, undefined, 'NONE');
+
+						//  OEB Logo
```

Comment:
Maybe instead of just commenting the different sections in this method, each section could have its own method? So something like `addOebLogo`, `addName` etc.. I believe this could make this huge method more readable